package.json

```diff
@@ -0,0 +1,51 @@
+{
+  "name": "check-status",
+  "version": "0.5.0",
+  "private": false,
+  "description": "Action to support workflows using the gitgitgadget tooling",
+  "main": "dist/index.js",
+  "scripts": {
+    "build": "ncc build src/miscRequest.ts -t -d -o dist",
+    "compile": "tsc",
+    "format": "prettier --write '**/*.ts'",
+    "format-check": "prettier --check **/*.ts",
+    "lint": "eslint src/**/*.ts",
+    "test:upr": "cmd /c test\\check-status.cmd update-open-prs",
+    "test:ucm": "cmd /c test\\check-status.cmd update-commit-mappings",
+    "test:hop": "cmd /c test\\check-status.cmd handle-open-prs",
+    "test:hnm": "cmd /c test\\check-status.cmd handle-new-mails",
+    "build:readme": "docker run -it --rm --name readme-gen -v %cd%:/usr/src/app -w /usr/src/app -e npm_package_version=%npm_package_version% node:18 npx --yes --trace-warnings github-action-readme-generator@1.0.12"
+  },
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/gitgitgadget/check-status.git"
+  },
+  "keywords": [
+    "slash",
+    "command",
+    "dispatch"
+  ],
+  "author": "Chris. Webster",
+  "license": "MIT",
+  "bugs": {
+    "url": "https://github.com/gitgitgadget/check-status/issues"
+  },
+  "homepage": "https://github.com/gitgitgadget/check-status#readme",
+  "dependencies": {
+    "@actions/core": "^1.9.1",
+    "@actions/github": "^5.0.3",
+    "find-git-exec": "^0.0.4"
+  },
+  "devDependencies": {
+    "@types/node": "^17.0.45",
+    "@typescript-eslint/parser": "^5.37.0",
+    "@vercel/ncc": "^0.34.0",
+    "eslint": "^8.23.1",
+    "eslint-plugin-anti-trojan-source": "^1.1.0",
+    "eslint-plugin-github": "^4.3.7",
+    "gitgitgadget": "file:../gitgitgadget/gitgitgadget-1.0.0.tgz",
```

Comment:
How about specifying it [directly](https://docs.npmjs.com/cli/v8/configuring-npm/package-json#github-urls), i.e. as a link to the repository?

Reply:
I was avoiding checking in the package as part of gitgitgadget.  Assumed there would be very little need for it.  The instructions identify how to create it.  It can be added to ggg if that is preferred.  The package could also be published to npm, etc. if that is a preferred direction.  Trying to limit the spread on something with limited use.

OTOH, did I misunderstand your suggestion?

Reply:
I thought that we could simply refer to the `gitgitgadget/gitgitgadget` repository without changing anything in that repository, is that not the case? Would we have to commit transpiled code?

Reply:
According to https://docs.npmjs.com/cli/v8/configuring-npm/package-json#git-urls-as-dependencies:

> When installing from a git repository, the presence of certain fields in the package.json will cause npm to believe it needs to perform a build.
> [...]
> build
> [...]
> install

I therefore _think_ that it should simply work, no changes in `gitgitgadget/gitgitgadget` required.

Reply:
What is needed to build the action are the gitgitgadget lib source files.  The `npm pack` makes them available with the deps info.  Having npm run a build is not going to make those files available as far as I can tell (no, I did not run a test).  The npm build feature seems like a way to make tools available that have not been published to a package manager.

Reply:
> The npm build feature seems like a way to make tools available that have not been published to a package manager.

Yes, and I think that is our use case. I should have a little bit of time to play with this next week.

Reply:
> Yes, and I think that is our use case.

Well, sort of.  From the doc it appears npm will do a build that we don't need.  It then does the pack (or equivalent) and installs that.  I was concerned it would try to use the build output but someone (okay, me) added `files` to the `package.json` and it all works.  I will update the `package.json` and `README.md` to show this simpler process.

Thanks for catching this and I guess I should have run that test.